bl/routefunctions.go

```diff
@@ -0,0 +1,129 @@
+package bl
+
+import (
+	"encoding/json"
+	"github.com/gin-gonic/gin"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"test/dal"
+)
+
+func GetBookByID(c *gin.Context){
+	username    := c.DefaultQuery("username","no_value")
+	id := c.Param("id")
+	client := dal.ElasticInitClient()
+	get1 := dal.ElasticGetBookByID(client, id)
+	updateUserData(username,"GET " +c.Request.RequestURI)
+	c.JSON(http.StatusOK, get1.Source)
+}
+
+func PutBook(c *gin.Context){
+	username    := c.DefaultQuery("username","no_value")
+	jsonData,err := ioutil.ReadAll(c.Request.Body)
+	if err!=nil{
+		panic(err)
+	}
+	book := dal.Book{}
+	json.Unmarshal(jsonData,&book)
+	client := dal.ElasticInitClient()
+	put1 := dal.ElasticPutBook(client, &book)
+	updateUserData(username,"PUT " +c.Request.RequestURI)
+	c.JSON(http.StatusOK, put1.Id)
+}
+
+func UpdateBook(c *gin.Context){
+	username    := c.DefaultQuery("username","no_value")
+	jsonData,err := ioutil.ReadAll(c.Request.Body)
+	if err!=nil{
+		panic(err)
+	}
+	book := dal.Book{}
+	json.Unmarshal(jsonData,&book)
+	client := dal.ElasticInitClient()
+	response := dal.ElasticUpdateBook(client,book.Title,c.Param("id"))
+	updateUserData(username,"POST " +c.Request.RequestURI)
+	c.JSON(http.StatusOK, response.Id)
+}
+
+func DeleteBook(c *gin.Context){
+	username    := c.DefaultQuery("username","no_value")
+	id := c.Param("id")
+	client := dal.ElasticInitClient()
+	response := dal.ElasticDeleteBook(client, id)
+	updateUserData(username,"Delete " +c.Request.RequestURI)
+	c.JSON(http.StatusOK,response)
+}
+
+func SearchBook(c *gin.Context){
+	username    := c.DefaultQuery("username","no_value")
+	title       := c.DefaultQuery("title","no_value")
+	authorName  := c.DefaultQuery("author_name","no_value")
+	priceRangeS := c.DefaultQuery("price_range", "no_value")
+	client      := dal.ElasticInitClient()
+	price_range := getPriceRange(priceRangeS)
+	var opcode int = 0
```

Comment:
```suggestion
	opcode := 0
```

`opcode` -stands for,? I don't understand the name..

Reply:
operation code. you can see elastic.go file in search book function for different numbers and cases.
opcode 0 - search with no arguments
opcode 1 - inly price range specified
...